src/background/feeder/local-storage.feeder.ts

```diff
@@ -0,0 +1,75 @@
+import browser from 'webextension-polyfill'
+import { ILocalStorageMessage } from '../../utils/message/local-storage'
+import { InterceptorReqMessageFormat, ResponseMessageFormat } from '../../utils/message/message-handler'
+import { DappSessionManager } from '../dapp-session.manager'
+import { isInternalMessage } from '../utils'
+
+export class LocalStorageFeeder {
+  constructor(private manager: DappSessionManager) {
+    this.serveEvents()
+  }
+  serveEvents(): void {
+    console.log('Register LocalStorageFeeder event listeners...')
+
+    browser.runtime.onMessage.addListener(
+      (message: InterceptorReqMessageFormat, sender: chrome.runtime.MessageSender) => {
+        if (!isInternalMessage(sender)) return
+
+        const { sessionId, key } = message
+
+        if (!sessionId || !this.manager.isValidSession(sessionId, sender)) return
+
+        const response: ResponseMessageFormat = {
+          key,
+          sender: 'background',
+          target: 'content',
+        }
+
+        if (this.isSetItemRequest(message)) {
+          const { payload } = message
+          console.log(`LocalStorageFeeder: store set -> ${payload}`)
+
+          if (!payload || !payload[0] || !payload[1]) {
+            response.error = `LocalStorageFeeder: wrong payload: Got ${payload}`
+
+            return response
+          }
+
+          return new Promise<ResponseMessageFormat>(resolve => {
+            this.manager.setStorageItem(sessionId, payload[0], payload[1]).then(() => {
+              resolve(response)
+            })
+          })
```

Comment:
I'd suggest extracting this as a function.

Reply:
how did you imagine this?

I need to alter `response` object of this function based on the validation/return value. What justifies this?

Reply:
Readability and more understandable code. There are many things going on in the function, and it is a bit long IMO, but the skeleton is simple:

```
if isSetItem
  setItem()
if isGetItem
  getItem()
```

So I'd extract the separate parts to keep every function small and more focused on a single thing.

I'd pass the `response` to the functions and alter it there, similarly to how middleware fns look.